File: medpay-web/src/types/hospital.ts
import type { HospitalStatus, SubscriptionTier } from './enums';

export interface HospitalCreateRequest {
  name: string;
  code: string;
  licenseNumber?: string;
  address?: string;
  city?: string;
  province?: string;
  contactPhone?: string;
  contactEmail?: string;
  logoUrl?: string;
  timezone?: string;
}

export interface HospitalUpdateRequest {
  name?: string;
  licenseNumber?: string;
  address?: string;
  city?: string;
  province?: string;
  contactPhone?: string;
  contactEmail?: string;
  logoUrl?: string;
  timezone?: string;
}

export interface HospitalResponse {
  id: string;
  name: string;
  code: string;
  licenseNumber: string;
  address: string;
  city: string;
  province: string;
  contactPhone: string;
  contactEmail: string;
  logoUrl: string;
  status: HospitalStatus;
  subscriptionTier: SubscriptionTier;
  configJson: Record<string, unknown>;
  timezone: string;
  createdAt: string;
  updatedAt: string;
}

export interface DepartmentRequest {
  name: string;
  code: string;
  parentId?: string;
  description?: string;
  sortOrder?: number;
}

export interface DepartmentResponse {
  id: string;
  name: string;
  code: string;
  parentId: string | null;
  description: string;
  sortOrder: number;
  status: string;
  children: DepartmentResponse[];
}

File: medpay-hospital/src/main/java/com/medpay/hospital/domain/Hospital.java
package com.medpay.hospital.domain;

import com.medpay.common.domain.BaseEntity;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;

@Entity
@Table(name = "hospital")
@Getter
@Setter
@NoArgsConstructor
public class Hospital extends BaseEntity {

    @Column(name = "name", nullable = false, length = 200)
    private String name;

    @Column(name = "code", nullable = false, unique = true, length = 50)
    private String code;

    @Column(name = "license_number", length = 100)
    private String licenseNumber;

    @Column(name = "address", columnDefinition = "TEXT")
    private String address;

    @Column(name = "city", length = 100)
    private String city;

    @Column(name = "province", length = 100)
    private String province;

    @Column(name = "contact_phone", length = 30)
    private String contactPhone;

    @Column(name = "contact_email", length = 200)
    private String contactEmail;

    @Column(name = "logo_url", length = 500)
    private String logoUrl;

    @Column(name = "status", nullable = false, length = 20)
    private String status = "ACTIVE";

    @Column(name = "subscription_tier", length = 30)
    private String subscriptionTier = "STANDARD";

    @JdbcTypeCode(SqlTypes.JSON)
    @Column(name = "config_json", columnDefinition = "jsonb")
    private String configJson = "{}";

    @Column(name = "timezone", length = 50)
    private String timezone = "Asia/Shanghai";
}
